sprint_17

Gvozdkov opened the pull request.

So1ovyov commented on Tracker/DaysValueTransformer.swift:
```diff
@@ -0,0 +1,60 @@
+    }
+}
+
+//@objc
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on Tracker/Helpers/UIcolor + UIFont.swift:
```diff
@@ -1,8 +1,46 @@
 
-import Foundation
+extension UIColor {
+//    static var backgroundDay: UIColor { UIColor(named: "BackgroundDay") ?? UIColor.lightGray }
```
Закомментированный код также и здесь можно удалить

So1ovyov commented on Tracker/OnboardingViewController/OnboardingViewController.swift:
```diff
@@ -0,0 +1,138 @@
+    private func constraintsSettingsView() {
+        view.addSubview(onboardingCollection)
+
+        view.addSubview(bigHeadline)
```
Обрати внимание на инструмент `forEach` при верстке и настройке элементов, он поможет тебе сократить повторяющийся код
Вот пример:
```
        [view1, 
         view2,
         view3].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
             view.addSubview($0)
        }
```
Таким образом все строки `translatesAutoresizingMaskIntoConstraints` и `addSubview` выше можно удалить и очень сильно сократить код

So1ovyov commented on Tracker/TrackerViewController/CreatingTracker/NewTrackerViewController/CategoryViewController/CategoryViewController.swift:
```diff
@@ -0,0 +1,186 @@
+                cell.clikerNewCategory(nil)
+            }
+            cell.backgroundColor = .clear
+//            cell.layer.cornerRadius = 16
```
Комментарий в коде

So1ovyov commented on Tracker/TrackerViewController/CreatingTracker/NewTrackerViewController/NewTrackerViewController.swift:
```diff
@@ -1,8 +1,80 @@
+
+final class NewTrackerViewController: UIViewController {
+    let newHabitViewController = NewHabitViewController()
+//    let unregulatedEventViewController = UnregulatedEventViewController()
```
Подчистить бы

So1ovyov commented on Tracker/TrackerViewController/CreatingTracker/NewTrackerViewController/NewTrackerViewController.swift:
```diff
@@ -1,8 +1,80 @@
+    
+    weak var delegate: NewHabitDelegate?
+    
+    lazy private var headingLabel: UILabel = {
```
Где-то у тебя `private lazy` а где-то `lazy private` давай придерживаться одного стиля и использовать последовательность общепринятую:
`private lazy var`

Commit:
Есть скриншот-тесты для главного экрана.

Commit:
В проекте подключена библиотека AppMetrica через менеджер CocoaPods.
Действия на главной отправляют:
открытие,
закрытие,
тап на кнопке добавления трека,
тап на треке,
тап на кнопке фильтра,
выбор редактирования в контекстном меню,
выбор удаления в контекстном меню.

Commit:
В проекте подключена библиотека AppMetrica через менеджер CocoaPods.

Commit:
Настроена анимация выделения ячейки как в макете

Commit:
Реализовано добавление экрана EditTrackerViewController для редактирования трекера. Добавлено переключение на данный экран при нажатии на ячейку и выборе опции 'Редактировать' в меню на экране TrackerViewController.

Commit:
Экран Онбординг соответствует макету

Commit:
Обновлены заглушки на экранах:
1) TrackerViewController
2) CategoryViewController

Commit:
Произведена локализация:
1) Экран FiltersViewController
2) Кнопка filtersButton на экране TrackerViewController

Commit:
Внесены следующие изменения:
1. По нажатию на ячейку с трекером появляется контекстное меню с пунктами:
- закрепить (закрепляет и открепляет ячейку)
- редактировать (запускает флоу с редактированием ячейки)
- удалить (удаляет ячейку с тренером)
2. Таблица с привычками теперь увеличивается в зависимости от количества привычек, а не фиксированного разрешения.
3. Добавлена метрика:
- edit для тапа на кнопку редактировать в контекстном меню
- delete для тапа на кнопку удалить в контекстном меню

Commit:
Исправлен баг когда на английской локализации трекеры не отображались

Commit:
Сверстан Launch Screen

Commit:
SearchBar теперь выполняет поиск по выбранному дню в datePicker, а не по всем дням, как это было ранее.

Commit:
Исправлен баг с отображение заглушки при поиске через searchBar
